### DISCUSSION TITLE
a way to change sortComparer in entityAdapter?

### DISCUSSION THREAD

**Discussion by jaykay-p** [reduxjs/redux-toolkit#1356] [Q&A] [answered]
Hi! :)
I am trying out entityAdapter, and want to dynamically change sortComparer.
Heres what I tried, but didnt worked out.
```ts
const adapter = createEntityAdapter({
  selectId: (itemList) => itemList.id,
  sortComparer: (a, b) => a.createdAt - b.createdAt;
});

const slice = createSlice({
  name: 'test',
  initialState: adapter.getInitialState(),
  reducers: {
    changeOrder: (state, action) => {
      chatInfosAdapter.sortComparer = (a, b) => b.createdAt - a.createdAt;
    },
  },
}
```

I wanted to test this before I apply more complex sort comparer logic in entityAdapter.
What will be the right way to change sort comparer?

btw, i am also trying out rtk-query and loving it so far! cant wait for ssr support 🙏 
Thanks!

**Comment by markerikson** [comment, marked as answer]
Not possible at the moment, sorry.

`createEntityAdapter` creates a closure, and there's currently no way to change the sorting behavior after creation.

You could in theory create more than one adapter with _different_ sorting methods, and switch which one you're using:

```js
if (someCondition) {
  sortByNameAdapter.upsertOne(state, action)
} else {
  sortByDateAdapter.upsertOne(state, action)
}
```
